src/components/Checkout/BillingInfo/BillingInfo.js

```diff
@@ -11,102 +12,158 @@ const CN = 'billing-info';
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <p>country</p>
-        <Dropdown
-          menuOptions={dropdownOptions.billingInfo.country}
-          value="Choose country..."
-          onDropdownChange={onDropdownChange}
-          className={`${CN}__dropdown`}
-        />
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <label htmlFor="city">city</label>
-        <input type="text" name={CN} className={`${CN}__input`} />
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <p>state</p>
-        <Dropdown
-          menuOptions={dropdownOptions.billingInfo.state}
-          value="Choose state..."
-          onDropdownChange={onDropdownChange}
-          className={`${CN}__dropdown`}
-        />
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <label htmlFor="zipcode">zip / postal code*</label>
-        <input type="number" name={CN} className={`${CN}__input`} />
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <label htmlFor="phone">phone</label>
-        <input type="number" name={CN} className={`${CN}__input`} />
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <label htmlFor="fax">fax</label>
-        <input type="number" name={CN} className={`${CN}__input`} />
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <label htmlFor="password">password*</label>
-        <input type="password" name={CN} className={`${CN}__input`} />
-      </div>
-      <div className={`${CN}__input-wrapper`}>
-        <label htmlFor="password">confirm password*</label>
-        <input type="password" name={CN} className={`${CN}__input`} />
-      </div>
-    </form>
-    <form className={`${CN}__radiobtns-form`}>
-      <div className={`${CN}__radiobtn`}>
-        <input
-          className={`${CN}__radiobtn-input`}
-          type="radio"
-          name="ship-to-this-address"
-          value="ship-to-this-address"
-        />
-        <label htmlFor="ship-to-this-address">Ship to this address</label>
-      </div>
-      <div className={`${CN}__radiobtn`}>
-        <input
-          className={`${CN}__radiobtn-input`}
-          type="radio"
-          name="ship-to-this-address"
-          value="ship-to-different-address"
-        />
-        <label htmlFor="guest">Ship to a different address</label>
-      </div>
-    </form>
-    <div className={`${CN}__checkbox-wrapper`}>
-      <input className={`${CN}__radiobtn-input`} type="checkbox" />
+class BillingInfo extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.form = React.createRef();
+    this.validate = this.validate.bind(this);
+  }
+
+  validate() {
+    const { showMessage } = this.props;
+
+    if (!this.form.current.reportValidity()) {
+      showMessage(VALIDATION_FAILED);
+      return this.form.current.reportValidity();
+    }
+
+    return this.form.current.reportValidity();
```

Comment:
You can move this call to the separate variable to call this only once. Plus line 28 is redundant.